File: app/src/main/java/com/ne1c/gitteroid/ui/activities/BaseActivity.java
package com.ne1c.gitteroid.ui.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.ne1c.gitteroid.Application;
import com.ne1c.gitteroid.di.components.ApplicationComponent;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        initDiComponent();
        super.onCreate(savedInstanceState);
    }

    protected abstract void initDiComponent();

    protected ApplicationComponent getAppComponent() {
        return ((Application) getApplication()).getComponent();
    }
}

File: app/src/main/kotlin/com/ne1c/gitteroid/ui/activities/BaseActivity.kt
package com.ne1c.gitteroid.ui.activities

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import com.ne1c.gitteroid.GitteroidApplication
import com.ne1c.gitteroid.di.components.ApplicationComponent

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        initDiComponent()
        super.onCreate(savedInstanceState)
    }

    protected abstract fun initDiComponent()

    protected val appComponent: ApplicationComponent
        get() = (application as GitteroidApplication).component
}
